src/modules/sync/diff/combinators.ts

```diff
@@ -52,11 +52,16 @@ export const makeLeafObjectHandler = <Entity extends object>(
   transformBeforeReplace: (v: Entity) => unknown = x => x,
 ): Handler<Entity> =>
 (sourceValue, targetValue) => {
-  const shouldReplace = zip(
-    Object.keys(sourceValue) as (keyof typeof customComparers)[],
-    zip(Object.values(sourceValue), Object.values(targetValue)),
-  )
-    .some(([key, [source, target]]) => !(customComparers[key]?.(source, target) ?? source === target));
+  const keys = new Set([
+    ...Object.keys(sourceValue),
+    ...Object.keys(targetValue),
+    ...Object.keys(customComparers),
+  ]) as Set<keyof Entity>;
+
+  const shouldReplace = Array.from(keys)
+    .some(key =>
+      !(customComparers[key]?.(sourceValue[key], targetValue[key]) ?? sourceValue[key] === targetValue[key])
```

Comment:
I know that it was there before, but is this part `sourceValue[key] === targetValue[key]` in the expression correct? I would think that we want to replace the object if any of the properties are different, meaning this expression instead `sourceValue[key] !== targetValue[key]`. Am I missing something? Can you write a test case for this scenario if it is indeed a bug?

Reply:
Oh, I didn't notice the negation is around the whole expression. Everything is alright then.